### ISSUE TITLE
Support .well-known to autodiscover backend/api and proxy urls

### ISSUE THREAD

**Issue by PureTryOut** [TeamPiped/Piped#3054]
### Describe the feature

A client can choose to let the user choose themselves which instance they want to use, e.g. KDE's [Plasmatube](https://invent.kde.org/multimedia/plasmatube) does this. However to make that work they have to enter the API URL rather than the frontend URL they know from using Piped in the browser. This API URL could literally be anything and there is no way for the user to know about it rather than opening devtools in their browser, which an average user wouldn't do.

It would be great if clients could instead just ask the user to enter the frontend URL (e.g. https://piped.video) rather than the API URL (https://pipedapi.kavin.rocks) and the client would be able to auto-detect the API URL and proxy URL through a system like .well-known (e.g. `https://piped.video/.well-known/piped`) where the used backend and proxy URL's are specified. That would even solve #3049 if the Piped web frontend by default looked for it's own base URL + `.well-known/piped`.

### Why would this be useful to add?

Clients will be able to auto-detect the API and proxy service to use while letting the user just care about the main instance URL. Users won't have to manually figure out what API URL they need for the instance they are using.

### Acknowledgements

- [X] I have searched the existing issues and this is **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this form.

**Comment by Bnyro** [comment]
As already explained in other issues, it's possible (and also very common) that some admins host an instance without a frontend (only using backend and proxy) for usage via 3rd party clients such as LibreTube. Apart from that instance frontends are not rendered on server side and thus there's no reliable way to match a frontend to a backend.

**Comment by PureTryOut** [comment]
> host an instance without a frontend (only using backend and proxy) 

Makes sense. I'm just trying to think of user-friendly ways of letting users use alternative clients and the web frontend without having to understand that there is a different backend URL in play. LibreTube hardcodes a few instances from what I can see but if you add a custom instance it asks for 2 separate URLs. Can we just think of ways to remedy that rather than closing this issue?

**Comment by Bnyro** [comment]
The frontend URL is only used for sharing videos in LibreTube.

Still, Piped will never require a frontend for hosting.

**Comment by PureTryOut** [comment]
Sure, but can we try to make this user-friendly somehow? :sweat_smile: 

Or I guess we can stick to the list on https://github.com/TeamPiped/Piped/wiki/Instances (have clients parse that) and for any other custom instance admins just have to tell their users what the URL is...
